docs/20_smart-contracts/40_guides/10_resources_guide.md

```diff
@@ -0,0 +1,232 @@
+
+#### Buy RAM In EOS
+
+For example the below command buys for account `bob` 0.1 EOS worth of RAM at the current market RAM price. The cost for the RAM and the execution of this transaction is covered by the `alice` account and the transaction is authorized by the `active` key of the `alice` account.
+
+```shell
+dune -- cleos system buyram alice bob "0.1 EOS" -p alice@active
+```
+
+#### Buy RAM In Bytes
+
+For example the below command buys for account `bob` 1000 RAM bytes at the current market RAM price. The cost for the RAM and the execution of this transaction is covered by the `alice` account and the transaction is authorized by the `active` key of the `alice` account.
+
+```shell
+dune -- cleos system buyrambytes alice bob "1000" -p alice@active
+```
+
+#### Buy RAM With EOS Wallet
+
+Another way to buy RAM is through an EOS wallet that supports this feature.
+
+### How Is RAM Calculated
+
+The necessary RAM needed for a smart contract to store its data is calculated from the used blockchain state.
+
+As a developer, to understand the amount of RAM your smart contract needs, you have to pay attention to the data structure underlying the multi-index tables your smart contract instantiates and uses. The data structure underlying one multi-index table defines a row in the table. And each data member of the data structure corresponds with a row cell of the table.
+To approximate the amount of RAM one multi-index row needs to be stored on the blockchain you have to add the size of the type of each data member and the memory overheads, defined by EOS code for various structures. Moreover, you must also consider if there are any indexes defined on your multi-index table. Find below the overheads defined by the EOS code for multi-index tables, indexes and data types:
+
+* [Multi-index RAM bytes overhead](https://github.com/AntelopeIO/leap/blob/f6643e434e8dc304bba742422dd036a6fbc1f039/libraries/chain/include/eosio/chain/contract_table_objects.hpp#L240)
+* [Overhead per row per index RAM bytes](https://github.com/AntelopeIO/leap/blob/a4c29608472dd195d36d732052784aadc3a779cb/libraries/chain/include/eosio/chain/config.hpp#L109)
+* [Fixed overhead shared vector RAM bytes](https://github.com/AntelopeIO/leap/blob/a4c29608472dd195d36d732052784aadc3a779cb/libraries/chain/include/eosio/chain/config.hpp#L108)
+* [Overhead per account RAM bytes](https://github.com/AntelopeIO/leap/blob/a4c29608472dd195d36d732052784aadc3a779cb/libraries/chain/include/eosio/chain/config.hpp#L110)
+* [Setcode RAM bytes multiplier](https://github.com/AntelopeIO/leap/blob/a4c29608472dd195d36d732052784aadc3a779cb/libraries/chain/include/eosio/chain/config.hpp#L111)
+* [RAM usage update function](https://github.com/AntelopeIO/leap/blob/9f0679bd0a42d6c24a966bb79de6d8c0591872a5/libraries/chain/apply_context.cpp#L725)
+
+### Related Documentation
```

Comment:
Let's remove this since we have a full smart contract guide coming next which will supersede these references